woocommerce/woocommerce-ios
# Add Copiable and Fakeable support to Hardware framework

The main task for #6683 


### Description


To simplify unit tests that use objects in the Hardware framework, this PR added Copiable and Fakeable support to Hardware.framework by adding the yaml files and running `rake generate` locally. I also updated the documentation for Copiable and Fakeable to include the instructions on adding support to a new Xcode framework.

To utilize the Copiable and Fakeable support, this PR also updated `CardBrand`, `CardPresentTransactionDetails`, `Charge`, `ChargeStatus`, `PaymentIntent`, `PaymentIntentStatus`, `PaymentMethod` in the Hardware framework to conform to Copiable and/or Fakeable protocols so that we can remove the lengthy `MockPaymentIntent` in `WooCommerceTests` and be more clear about the necessary properties to mock in a test case.

There is still a `MockPaymentIntent` in `YosemiteTests`, but the replacement includes many more diffs so I'll make those changes in a follow-up PR.

### Testing instructions

The changes should only affect testing, but feel free to run the app with IPP as a confidence check!

---
